launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -230,17 +230,11 @@ String getScalaVersion() {
       return scala;
     }
     String sparkHome = getSparkHome();
-    File scala212 = new File(sparkHome, "launcher/target/scala-2.12");
     File scala213 = new File(sparkHome, "launcher/target/scala-2.13");
-    checkState(!scala212.isDirectory() || !scala213.isDirectory(),
+    checkState(!scala213.isDirectory(),
```

Comment:
I'd comment out all this code instead, to save it in case we want to bring it back when Scala 3 is supported.
But you just return "2.13" from this method for now unconditionally.

Reply:
Is it not necessary to check `scala213.isDirectory()`?

Reply:
OK

Reply:
done